value.go

```diff
@@ -194,9 +199,21 @@ func (v *MrbValue) Hash() *Hash {
 
 // String returns the "to_s" result of this value.
 func (v *MrbValue) String() string {
-	value := C.mrb_obj_as_string(v.state, v.value)
-	result := C.GoString(C.mrb_string_value_ptr(v.state, value))
-	return result
+	if v.IsDead() {
+		return ""
```

Comment:
ok, I'm not so sure about this myself. there's no way to determine between a dead value and a regular string in this instance, which is valuable data that we both want to preserve, jsut in different ways.

What if we made it return `(string, error)` instead? We could tag current master with a version to distinguish the API change, if you wanted.

There might be a few more spots like this too.